posthog/batch_exports/service.py

```diff
@@ -100,14 +129,47 @@ async def pause_schedule(temporal: Client, schedule_id: str, note: str | None =
     await handle.pause(note=note)
 
 
-def unpause_batch_export(temporal: Client, batch_export_id: str, note: str | None = None) -> None:
+def unpause_batch_export(
+    temporal: Client, batch_export_id: str, note: str | None = None, backfill: bool = False
+) -> None:
     """Pause this BatchExport.
 
-    We pass the call to the underlying BatchExportSchedule. This exists here as a convinience so that users only
-    need to interact with a BatchExport.
+    We pass the call to the underlying Temporal Schedule. Additionally, we can trigger a backfill
+    to backfill runs missed while paused.
+
+    Args:
+        temporal: The Temporal client to execute calls.
+        batch_export_id: The ID of the BatchExport to unpause.
+        note: An optional note to include in the Schedule when unpausing.
+        backfill: If True, a backfill will be triggered since the BatchExport's last_paused_at.
+
+    Raises:
+        BatchExportIdError: If the provided batch_export_id doesn't point to an existing BatchExport.
     """
-    BatchExport.objects.filter(id=batch_export_id).update(paused=False)
-    unpause_schedule(temporal, schedule_id=batch_export_id, note=note)
+    batch_export = BatchExport.objects.get(id=batch_export_id)
+
+    if not batch_export:
+        raise BatchExportIdError(batch_export_id)
+
+    if batch_export.paused is False:
+        return
+
+    try:
+        unpause_schedule(temporal, schedule_id=batch_export_id, note=note)
+    except Exception as exc:
+        raise BatchExportServiceRPCError(f"BatchExport {batch_export_id} could not be unpaused") from exc
+
+    batch_export.paused = False
+    batch_export.save()
+
+    if backfill is False:
+        return
+
+    batch_export.refresh_from_db()
+    start_at = batch_export.last_paused_at
+    end_at = batch_export.last_updated_at
+
+    backfill_export(temporal, batch_export_id, start_at, end_at)
```

Comment:
Is there any chance that the PG db change can go through and then this temporal api calls fail leaving us in an odd state? I guess people can just pause and unpause again to trigger a retry?

Reply:
Yeah, good call 👍 , I can wrap this in a try/finally.

Reply:
Wait, I don't see what odd state we would be in: if the Temporal backfill call fails, just retry the backfill again? The db state is consistent as we did unpause.

Reply:
You do lose the ability to backfill automatically for the period you were paused, but you can pass that manually, and we can have an option in the frontend like "Backfill last paused period" that gets the `last_paused_at` and does it for you.

Reply:
yup